File: U2-Toolkit/.NET Samples/samples/VB.NET/UniVerse/EntityFramework_CodeFirst2/Customer.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Entity

Namespace EntityFramework_CodeFirst2
    Public Class Customer
        Sub New()
        End Sub


        <Key()> _
        Public Property CUSTID() As Integer

        Public Property FNAME() As String
        Public Property LNAME() As String
        Public Property FULLADDR() As String
    End Class

    Public Class CustomerContext
        Inherits DbContext
        Public Sub New()
        End Sub

        Public Property Customers() As DbSet(Of Customer)
        Protected Overrides Sub OnModelCreating(modelBuilder As DbModelBuilder)

            modelBuilder.Entity(Of Customer)().[Property](Function(s) s.FNAME).IsRequired()
        End Sub

    End Class
End Namespace

File: U2-Toolkit/.NET Samples/samples/C#/UniVerse/EntityFramework_CodeFirst2/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;

namespace EntityFramework_CodeFirst2
{
    public class Customer
    {
        Customer()
        {
        }

        [Key]
        public int CUSTID { get; set; }

        public string FNAME { get; set; }
        public string LNAME { get; set; }
        public string FULLADDR { get; set; }
    }

    public class CustomerContext : DbContext
    {
        public CustomerContext()
        {
        }
        public DbSet<Customer> Customers { get; set; }
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Customer>()
                .Property(s => s.FNAME)
                
                .IsRequired();
            modelBuilder.Conventions.Remove<IncludeMetadataConvention>(); 
        }
    }



}
